docs/source/conf.py

```diff
@@ -441,13 +452,68 @@ def check_timeout_and_api_kwargs_presence(obj: object) -> int:
     )
 
 
+def create_return_admonitions() -> dict[str, str]:
+    """Creates 'Returned in' admonitions for classes that are returned in Bot's methods."""
+
+    methods_for_class = defaultdict(list)
+
+    for method in [
+        m[0]
+        for m in inspect.getmembers(telegram.Bot, inspect.isfunction)  # not .ismethod
+        if not m[0].startswith("_") and m[0].islower()  # islower() to avoid camelCase methods
+    ]:
+        sig = inspect.signature(getattr(telegram.Bot, method))
+        return_type_string = str(sig.return_annotation)
+        if "telegram" not in return_type_string:
+            continue
+
+        class_names = (
+            return_type_string.removeprefix("typing.Union[")
+            .removeprefix("typing.Tuple[")
+            .removeprefix("<class '")
+            .removesuffix("'>")
+            .removesuffix("]")
+            .split(", ")  # for the cases like Union[telegram.Something, telegram.SomethingElse]
+        )
```

Comment:
actually let's make it an exact match instead of getting the `__name__` (i.e. `str(telegram_obj)`), and then in `autodoc_process_docstring`, we can do `class_name = str(obj)` (if `what == "class"`). 

We'll get full matches this way: 
``` python
>>> a = typing.get_args(typing.Union[telegram.BotCommand, True])
>>> str(a[0])
"<class 'telegram._botcommand.BotCommand'>"
>>> str(telegram.BotCommand)
"<class 'telegram._botcommand.BotCommand'>"
```

Reply:
Very good, wouldn't have figured it out myself.
Done